File: app/src/main/java/com/helloworld/faceinteract/Face.java
package com.helloworld.faceinteract;

import com.arcsoft.facedetection.AFD_FSDKFace;
import com.arcsoft.facerecognition.AFR_FSDKFace;

import java.io.StringWriter;
import java.util.ArrayList;
import java.util.List;

public class Face
{
    private String name;
    private List<AFR_FSDKFace> faces;
    public Face(String name)
    {
        this.name = name;
        faces = new ArrayList<>();
    }
    public Face(String name, AFR_FSDKFace sdkFace)
    {
        this(name);
        faces.add(sdkFace);
    }
    public Face(String name, AFD_FSDKFace detectionFace)
    {

    }
    public String getName()
    {
        return name;
    }
    public void setName(String name) { this.name = name; }
    public void addSdkFace(AFR_FSDKFace sdkFace)
    {
        faces.add(sdkFace);
    }
    public AFR_FSDKFace getFirstSdkFace()
    {
        if (faces.size() == 0)
        {
            return null;
        }
        return faces.get(0);
    }
}

File: app/src/main/java/com/helloworld/faceinteract/Face.kt
package com.helloworld.faceinteract

import com.arcsoft.facerecognition.AFR_FSDKFace

/**
 * Stores name of face and multiple face data.
 */
class Face(var name: String) {
    private val faces = arrayListOf<AFR_FSDKFace>()

    val firstSdkFace: AFR_FSDKFace?
        get() = try {
            faces.first()
        } catch (e: NoSuchElementException) {
            null
        }

    constructor(name: String, sdkFace: AFR_FSDKFace) : this(name) {
        faces.add(sdkFace)
    }

    fun addSdkFace(sdkFace: AFR_FSDKFace?) {
        if (sdkFace != null) faces.add(sdkFace)
    }
}
